zomes/chat/src/entries/message/handlers.rs

```diff
@@ -209,6 +209,22 @@ fn active_chatters(chatters_path: Path) -> ChatResult<(usize, Vec<AgentPubKey>)>
     Ok((total, active))
 }
 
+pub(crate) fn get_active_chatters() -> ChatResult<ActiveChatters> {
+    let me = agent_info()?.agent_latest_pubkey;
+    let chatters_path: Path = chatters_path();
+    let (_total, mut chatters) = active_chatters(chatters_path)?;
+    chatters.retain(|a| *a != me);
+    Ok(ActiveChatters { chatters })
+}
+
+pub(crate) fn signal_specific_chatters(input: SignalSpecificInput) -> ChatResult<()> {
+    remote_signal(
+        &SignalPayload::Message(input.signal_message_data),
+        input.chatters,
+    )?;
+    Ok(())
+}
```

Comment:
what's the reasoning behind having a separate `signal_specific_chatters` zome call, distinct from the `create_message` zome call?
In a traditional app, the functionality for sending notifications would typically be part of the `create` api call, you wouldn't have a separate call for that. Is there an `hc` specific reason to do it differently?

The alternative would be to pass the list of chatters to `create_message` and have it send the signal itself.

Reply:
Zome calls each hold a database transaction such that none of the commits from a zome call are actually committed until the whole thing completes successfully. If signaling happened in the same zome call as message creation then I see two potential issues with that: 

1. The signal will be sent out before the message is even in the actual source chain which means that it is valid network behavior to send a message signal without committing that message to your source chain.
2. If signaling fails for some reason, then the message will not be created even if that part was successful

Reply:
That's exactly it.  Create_message can fail.  We don't want to send the signals at until after it succeeds.  There's supposed to be a post-commit hook at some point, and when that happens we could put it in there.

Reply:
Makes sense, thanks.